frontend/public/module/k8s/types.ts

```diff
@@ -981,8 +981,34 @@ export type VolumeSnapshotKind = K8sResourceCommon & {
   };
   spec: {
     source: {
-      persistentVolumeClaimName: string;
+      persistentVolumeClaimName?: string;
+      volumeSnapshotContentName?: string;
     };
     volumeSnapshotClassName: string;
   };
 };
+
+export type VolumeSnapshotContentKind = K8sResourceCommon & {
+  status: {
+    readyToUse: boolean;
+    restoreSize?: number;
+    snapshotHandle?: string;
+    error?: {
+      message: string;
+      time: string;
+    };
+  };
```

Comment:
Let's create a common volume snapshot status type and use that. It will make the type declaration common between them.